### PR TITLE
Add Admin::DomainBlocks entity and methods

### LINKED ISSUE #322: Implement Admin Domain Blocks API
Mastodon's Admin Domain Blocks API allows an admin to disallow certain domains to federate. The goal of this task is to implement the following endpoints of this API in BigBone:

- List all blocked domains
- Get a single blocked domain
- Block a domain from federating
- Update a domain block
- Remove a domain block

The Admin Domain Blocks API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/admin/domain_blocks/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

This PR adds support for the Admin::DomainBlocks entity and methods.

> [!NOTE]
> The `Severity` class is already implemented in the non-Admin `DomainBlocks` entity but does not contain the `NOOP` variant.
> I decided to implement another one inside of `AdminDomainBlocks`. That’ll leave us with some duplication but will also ensure expected behaviour.

Closes #322.

# Type of Change

- New feature
- Documentation

# Breaking Changes

- None

# How Has This Been Tested?

I’ve added new unit tests for all methods

# Mandatory Checklist

- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [x] I have added KDoc documentation to all public methods

# Optional Things To Check

The items below are some more things to check before asking other people to review your code.

- [x] In case you worked on a new feature: Did you also implement the reactive endpoint (bigbone-rx)?
- [x] In case you added new `*Methods` classes: Did you also reference it in the `MastodonClient` main class?
- [x] Did you also update the documentation in the `/docs` folder (e.g. API Coverage page)?

### CODE REVIEW

File: bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt
Comment (PattaFeuFeu) [inline bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt:128]: Moved up to retain alphabetical order

**Review by andregasser — APPROVED** [review]
All good from my side! 👍

**Comment by PattaFeuFeu** [comment]
@andregasser You may need to approve again after the latest merge from main

**Comment by andregasser** [comment]
> @andregasser You may need to approve again after the latest merge from main

It still seems approved from my side. Build running. Feel free to go ahead  😊

**Comment by PattaFeuFeu** [comment]
@andregasser The button was greyed out before for me even though all build steps passed 🤔 Worked now though.

**Pull request merged**
